docs/markdown/Wrap-dependency-system-manual.md

```diff
@@ -136,25 +136,55 @@ thousands of lines of code. Once you have a working build definition,
 just zip up the Meson build files (and others you have changed) and
 put them somewhere where you can download them.
 
-Meson build patches are only supported for wrap-file mode. When using
-wrap-git, the repository must contain all Meson build definitions.
+Prior to *0.55.0* Meson build patches were only supported for wrap-file mode.
+When using wrap-git, the repository must contain all Meson build definitions.
+Since *0.55.0* Meson build patches are supported for any wrap modes, including
+wrap-git.
 
 ## `provide` section
 
 *Since *0.55.0*
 
 Wrap files can define the dependencies it provides in the `[provide]` section.
-When a wrap file provides the dependency `foo` any call do `dependency('foo')`
-will automatically fallback to that subproject even if no `fallback` keyword
-argument is given. It is recommended for subprojects to call
-`meson.override_dependency('foo', foo_dep)`, dependency name can then be added into
-the special `dependency_names` entry which takes comma separated list of dependency
-names. For backward compatibility with subprojects that does not call
-`meson.override_dependency()`, the variable name can be provided in the wrap file
-with entries in the format `dependency_name = variable_name`,
-where `dependency_name` usually match the corresponding pkg-config name and
-`variable_name` is the name of a variable defined in the subproject that should
-be returned for that dependency.
+
+```
+[provide]
+dependency_names = foo-1.0
```

Comment:
I think using examples with "foo-1.0" makes it harder to concentrate on the provided name semantics since a lot of pkgconfig library names don't follow that scheme.

Reply:
Hm, I made that change specifically because most pkgconfig libraries follows that scheme (glib-2.0, gstreamer-1.0, gtk+-3.0, ...) but that could be a bias because I'm from GNOME world?

Reply:
Looking at /usr/lib/x86_64-linux-gnu/pkgconfig/ you're actually right, most don't... I'm biaised :D

But still, if your pkgconfig file is named `foo` and you have `foo.wrap` you don't even need the `[provide]` section, it automatically fallback to a wrap with the same name as your dependency.

Reply:
maybe use foo-glib as example dependency that would need to be added explicitly?

Reply:
Hm, I've been using foo-1.0 across the whole text, I don't think foo-glib makes much difference. IMHO `foo-1.0` is common enough for a dependency name that does not match the project name.